code/components/discord/src/DiscordPresence.cpp

```diff
@@ -108,4 +158,40 @@ static InitFunction initFunction([]()
 
 		g_richPresenceChanged = true;
 	});
+
+	fx::ScriptEngine::RegisterNativeHandler("SET_RICH_PRESENCE_ASSET", [](fx::ScriptContext& context)
+	{
+		const char* str = context.GetArgument<const char*>(0);
+
+		if (str)
+		{
+			g_richPresenceOverrideAsset = str;
+		}
+		else
+		{
+			g_richPresenceOverrideAsset = "";
+		}
+
+		g_richPresenceChanged = true;
+	});
+
+	fx::ScriptEngine::RegisterNativeHandler("SET_RICH_PRESENCE_ID", [](fx::ScriptContext& context)
+	{
+		const char* str = context.GetArgument<const char*>(0);
+
+		if (str)
+		{
+			DiscordId = str;
+		}
+		else
+		{
+			DiscordId = "382624125287399424";
+		}
+		Discord_Shutdown();
```

Comment:
Is this call actually safe? Also, probably should be called `SET_DISCORD_APPLICATION_ID` or so.

Reply:
according to discord's docs, it is, it terminates the thread allowing the application to shut down gracefully or discord rich presence to be reinitialized, without this, i am pretty sure the threads would just stack one onto another every time you changed the app id.